distributed/comm/tcp.py

```diff
@@ -335,6 +338,8 @@ class TLS(TCP):
     A TLS-specific version of TCP.
     """
 
+    max_shard_size = min(C_INT_MAX, TCP.max_shard_size)
```

Comment:
Would suggest adding a comment here to explain why this is needed. Something like the following

```suggestion
    # Workaround for OpenSSL 1.0.2 (can drop with OpenSSL 1.1.1)
    max_shard_size = min(C_INT_MAX, TCP.max_shard_size)
```